### PR TITLE
buildsys build-kit

### LINKED ISSUE #70: kits: buildsys build-kit command
## Description

In order to build a kit image, buildsys will need to be invoked from a kit's `build.rs` file with the command `buildsys build-kit`. This is a new subcommand for buildsys

## Definition of Done

For the sake of making smaller, incremental PRs, it is Ok if this is not fully functional at first. It will be made fully functional in https://github.com/bottlerocket-os/twoliter/issues/17

This issue is done when some semblance of the command exists in Buildsys.

### LINKED ISSUE #73: kits: integrate with Makefile.toml and Dockerfile
## Description

Integrate the `twoliter build kit` command into the Makefile.toml ecosystem. Make it possible to build a variant where the packages are built into a kit first. Possibly there should be a default mode that detects the absence of kits in the project and creates a single all-packages kit such that the Bottlerocket build still works without yet introducing Bottlerocket's kits (just an idea).

## Definition of Done

When `cargo make kit` works. And when `twoliter build kit` invokes the proper Makefilte.toml prerequisites before building the kit.

### PR DESCRIPTION
**Pull request by webern**
**Issue number:**

Closes #70 
Closes #73
Follows #226 

**Description of changes:**

Add a `build-kit` command that can create a local kit!

**Testing done:**

- [x] Build the test project kits, it worked!
- [x] Build Bottlerocket, it's not broken!

Kit test: (note `build-variant` fails because the test project does not have a kernel)

```
   Compiling pkg-a-1_27 v0.1.0 (/home/somebody/repos/twoliter/tests/projects/local-kit/packages/pkg-a-1.27)
   Compiling core-kit v0.1.0 (/home/somebody/repos/twoliter/tests/projects/local-kit/kits/core-kit)
   Compiling pkg-c v0.1.0 (/home/somebody/repos/twoliter/tests/projects/local-kit/packages/pkg-c)
   Compiling pkg-b v0.1.0 (/home/somebody/repos/twoliter/tests/projects/local-kit/packages/pkg-b)
   Compiling pkg-d v0.1.0 (/home/somebody/repos/twoliter/tests/projects/local-kit/packages/pkg-d)
   Compiling pkg-f v0.1.0 (/home/somebody/repos/twoliter/tests/projects/local-kit/packages/pkg-f)
   Compiling pkg-g v0.1.0 (/home/somebody/repos/twoliter/tests/projects/local-kit/packages/pkg-g)
   Compiling extra-1-kit v0.1.0 (/home/somebody/repos/twoliter/tests/projects/local-kit/kits/extra-1-kit)
   Compiling extra-2-kit v0.1.0 (/home/somebody/repos/twoliter/tests/projects/local-kit/kits/extra-2-kit)
   Compiling pkg-e v0.1.0 (/home/somebody/repos/twoliter/tests/projects/local-kit/packages/pkg-e)
   Compiling extra-3-kit v0.1.0 (/home/somebody/repos/twoliter/tests/projects/local-kit/kits/extra-3-kit)
```

```
$ tree rpms kits
rpms
├── pkg-a-1.27
│   └── bottlerocket-pkg-a-0.0-0.x86_64.rpm
├── pkg-b
│   └── bottlerocket-pkg-b-0.0-0.x86_64.rpm
├── pkg-c
│   └── bottlerocket-pkg-c-0.0-0.x86_64.rpm
├── pkg-d
│   └── bottlerocket-pkg-d-0.0-0.x86_64.rpm
├── pkg-e
│   └── bottlerocket-pkg-e-0.0-0.x86_64.rpm
├── pkg-f
│   └── bottlerocket-pkg-f-0.0-0.x86_64.rpm
└── pkg-g
    └── bottlerocket-pkg-g-0.0-0.x86_64.rpm
kits
├── core-kit
│   └── x86_64
│       ├── Packages
│       │   └── bottlerocket-pkg-a-0.0-0.x86_64.rpm
│       └── repodata
│           ├── 1dc260ef8cb9d32636f49827fe04aaf7bac406c21f8d8cdaf1d7766089ebec90-filelists.xml.zst
│           ├── b1d71027e5090bb17bce76a9358bc5549cd3e2710022f5c94c45012882dc61c1-primary.xml.zst
│           ├── e6ad51b865bac427c57e6cc98fe6047c2165831834d252d6476faebbb0d2c6b5-other.xml.zst
│           └── repomd.xml
├── extra-1-kit
│   └── x86_64
│       ├── Packages
│       │   ├── bottlerocket-pkg-b-0.0-0.x86_64.rpm
│       │   └── bottlerocket-pkg-d-0.0-0.x86_64.rpm
│       └── repodata
│           ├── 27986a40a542036143227d3c95bd0d482cbcf5bdfa4c8a2c17319261a81133f3-filelists.xml.zst
│           ├── 2e9747b211b81b6d923b97b925eed64678fa0eb3abee546eba56a5916c2ecdaa-primary.xml.zst
│           ├── 4e63bbf66f02a170e5fdb959693d76074370b0725228b6c4ebcd3ebe780fd89b-other.xml.zst
│           └── repomd.xml
├── extra-2-kit
│   └── x86_64
│       ├── Packages
│       │   └── bottlerocket-pkg-c-0.0-0.x86_64.rpm
│       └── repodata
│           ├── 07f3159e9651db8be604dbb4eadf808789d7b831b1350899d0aed191ef46ddcd-other.xml.zst
│           ├── 6c649dcc73afbff1e2c1982e64e1acc224b123726de816c3435982703354dada-primary.xml.zst
│           ├── 9e3b96057582abd3c874a100efab6491ebd42cfa2c28c14ef5b7afe7d72595e1-filelists.xml.zst
│           └── repomd.xml
└── extra-3-kit
    └── x86_64
        ├── Packages
        │   ├── bottlerocket-pkg-e-0.0-0.x86_64.rpm
        │   ├── bottlerocket-pkg-f-0.0-0.x86_64.rpm
        │   └── bottlerocket-pkg-g-0.0-0.x86_64.rpm
        └── repodata
            ├── 230dd1c616e0f9a38df7664e1be7436190fdb143a7cabb9924a6b9593cd75421-other.xml.zst
            ├── 4ae4927f963e446f083b43590b33199a7aa54d1083152d9466b5e5fba566bbde-primary.xml.zst
            ├── ce0523598374330d2ee6a6fb9ac142ffa0ca1ed58eb92c3a83dc4562cb61d5c8-filelists.xml.zst
            └── repomd.xml
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by jmt-lab — APPROVED** [review]
LGTM

**Comment by webern** [comment]
> https://github.com/bottlerocket-os/twoliter/compare/60b99673a8ba4903c9044b3b5db31f824c4899a9..c6f4995a0349017e36b98c230e1076dec545d71d

Rebase

File: tools/buildsys/src/main.rs [hunk 1]
```diff
@@ -199,6 +202,24 @@ fn build_package(args: BuildPackageArgs) -> Result<()> {
...
+    let manifest_file = "Cargo.toml";
+    println!("cargo:rerun-if-changed={}", manifest_file);
+
+    let manifest = Manifest::new(
+        args.common.cargo_manifest_dir.join(manifest_file),
+        &args.common.cargo_metadata_path,
+    )
+    .context(error::ManifestParseSnafu)?;
+
+    supported_arch(manifest.info(), args.common.arch)?;
```
Comment (bcressey) [inline tools/buildsys/src/main.rs]: The `supported_arch` function does nothing for a non-variant manifest.

**Review by bcressey — APPROVED** [review]
LGTM, one minor issue.

**Comment by webern** [comment]
> https://github.com/bottlerocket-os/twoliter/compare/c6f4995a0349017e36b98c230e1076dec545d71d..601e2bdc86e2ed7aedd2fae468fa58b0b5bcfbb4

Rebase which appears to have included a conflict resolution error... now fixing.

**Comment by webern** [comment]
> https://github.com/bottlerocket-os/twoliter/compare/601e2bdc86e2ed7aedd2fae468fa58b0b5bcfbb4..75009f670146885547508155b354225bdf1fa769

Fix the rebase mistake and also add `KIT` to the rebuilds for `BUILDSYS_ARCH`.

**Comment by webern** [comment]
> https://github.com/bottlerocket-os/twoliter/compare/75009f670146885547508155b354225bdf1fa769..46d5bec8a1fa43eb3b952956a81156f72985e2e9

Address Ben's comment (don't use the supported_arch function).

**Pull request merged**